src/main/java/io/excaliburfrc/robot/subsystems/Drive.java

```diff
@@ -19,15 +43,112 @@
+        leftFollower.restoreFactoryDefaults(),
+        rightLeader.restoreFactoryDefaults(),
+        rightFollower.restoreFactoryDefaults(),
+        // set the motors to coast mode -- we don't want to break them!
+        leftLeader.setIdleMode(CANSparkMax.IdleMode.kBrake),
+        leftFollower.setIdleMode(CANSparkMax.IdleMode.kBrake),
+        rightLeader.setIdleMode(CANSparkMax.IdleMode.kBrake),
+        rightFollower.setIdleMode(CANSparkMax.IdleMode.kBrake),
+        // have the leader send its applied output as frequently as possible,
+        // to speed up follower response
+        leftLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus0, minimal_FRAME_PERIOD),
+        leftLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus1, minimal_FRAME_PERIOD),
+        leftLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus2, minimal_FRAME_PERIOD),
+        rightLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus0, minimal_FRAME_PERIOD),
+        rightLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus1, minimal_FRAME_PERIOD),
+        rightLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus2, minimal_FRAME_PERIOD),
+        // other status frames can be reduced to almost never
+        leftFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus0, MAXIMAL_FRAME_PERIOD),
+        leftFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus1, MAXIMAL_FRAME_PERIOD),
+        leftFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus2, MAXIMAL_FRAME_PERIOD),
+        rightFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus0, MAXIMAL_FRAME_PERIOD),
+        rightFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus1, MAXIMAL_FRAME_PERIOD),
+        rightFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus2, MAXIMAL_FRAME_PERIOD),
+        // setup following
+        leftFollower.follow(leftLeader),
+        rightFollower.follow(rightLeader));
+
+    odometry = new DifferentialDriveOdometry(ahrs.getRotation2d());
+  }
+
+  public void resetOdometry(Pose2d pose) {
+    leftEncoder.setPosition(0);
+    rightEncoder.setPosition(0);
+    offset = pose.getRotation();
+    odometry.resetPosition(pose, ahrs.getRotation2d());
+  }
+
+  public Pose2d getPose() {
+    return odometry.getPoseMeters();
+  }
+
+  public void achieveVelocity(double left, double right) {
+    leftController.setReference(
+        left, ControlType.kVelocity, 0, feedforward.calculate(left), ArbFFUnits.kVoltage);
+    rightController.setReference(
+        right, ControlType.kVelocity, 0, feedforward.calculate(right), ArbFFUnits.kVoltage);
   }
 
-  public Command arcadeDriveCommend(DoubleSupplier xSpeed, DoubleSupplier zRotation) {
+  public Command arcadeDriveCommand(DoubleSupplier xSpeed, DoubleSupplier zRotation) {
     return new RunCommand(
         () -> drive.arcadeDrive(xSpeed.getAsDouble(), zRotation.getAsDouble()), this);
   }
+
+  public void tankDrive(double left, double right) {
+    leftLeader.set(left);
+    rightLeader.set(right);
+    drive.feed();
+  }
```

Comment:
this function can use `drive.tankDrive(left, right)` if needs to be exposed, is completely redundant if not.
in any case, can be split into a more basic drivetrain PR